zap/src/test/java/org/zaproxy/zap/authentication/AuthenticationMethodPollUrlUnitTest.java

```diff
@@ -279,4 +279,62 @@ void shouldPollOnSpecifiedNumberOfRequestsPerUser() throws NullPointerException,
         // Second poll for user2
         assertThat(orderedReqs.size(), is(4));
     }
+
+    @Test
+    void shouldHandlePollHeadersWithColonsInValues() throws Exception {
+        // Given
+        String test = "/shouldHandlePollHeadersWithColonsInValues/test";
+        String pollUrl = "/shouldHandlePollHeadersWithColonsInValues/pollUrl";
+        final List<String> orderedReqs = new ArrayList<>();
+        final List<HttpMessage> capturedPollMessages = new ArrayList<>();
+
+        setMessageHandler(
+                msg -> {
+                    String path = msg.getRequestHeader().getURI().getPath();
+                    if (pollUrl.equals(path)) {
+                        orderedReqs.add(path);
+                        capturedPollMessages.add(msg);
+                        msg.setResponseBody(LOGGED_IN_BODY);
+                    }
+                });
+
+        HttpMessage testMsg = this.getHttpMessage(test);
+        HttpMessage pollMsg = this.getHttpMessage(pollUrl);
+
+        method.setAuthCheckingStrategy(AuthCheckingStrategy.POLL_URL);
+        method.setPollUrl(pollMsg.getRequestHeader().getURI().toString());
+        method.setLoggedInIndicatorPattern(LOGGED_IN_INDICATOR);
+
+        // Set poll headers with colons in values (like JWT tokens, dates, etc.)
```

Comment:
The test name already indicates that we are testing headers with colons. The comment is also misleading, the Content-Type does not have colons in the value.